### PR TITLE
sev: update sev-snp kernel version

### PR DESCRIPTION
**Pull request by enterJazz**

### CODE REVIEW

**Comment by mmisono** [comment]
We can merge this after solving zfs compiling issue.

**Comment by Mic92** [comment]
There is https://github.com/openzfs/zfs/releases/tag/zfs-2.2.0-rc4
However we might need patches from master: https://github.com/openzfs/zfs/commits/master
So maybe we need to upgrade to zfs master here: https://github.com/TUM-DSE/nixpkgs/blob/f178b9f277eb47e7b86c06adc1c1e17964ad9f0e/pkgs/os-specific/linux/zfs/unstable.nix#L12
One can pass commits here: https://github.com/TUM-DSE/nixpkgs/blob/f178b9f277eb47e7b86c06adc1c1e17964ad9f0e/pkgs/os-specific/linux/zfs/generic.nix#L25

**Comment by Mic92** [comment]
https://github.com/TUM-DSE/nixpkgs/commit/58b09ee69c91b9e7a8a61c607a5ca89098c52019

File: pkgs/kernels/linux-sev-snp.nix [hunk 1]
```diff
@@ -80,6 +80,16 @@ let
     extraPatches = [ ];
   };
 
+  snp_latest = {
+    owner = "AMDESE";
+    repo = "linux";
+    rev = "ad9c0bf475ecde466a065af44fc94918f109c4c9";
+    sha256 = "sha256-cAgiVA7bKuFteF/GaklTm1M6SkDN61LpPBNhflN1M1o=";
+    version = "6.5";
+    modDirVersion = "6.5.0-rc2";
```
Comment (Mic92) [inline pkgs/kernels/linux-sev-snp.nix:89]:
```suggestion
    modDirVersionArg = "6.5.0-rc2";
```

**Pull request merged**